src/Sql/Sql/ManagedDatabase/Services/AzureSqlManagedDatabaseCommunicator.cs

```diff
@@ -178,6 +178,69 @@ public ManagedDatabaseRestoreDetailsResult GetLogReplayStatus(string resourceGro
                 databaseName);
         }
 
+        public void Move(string resourceGroupName, string managedInstanceName, string databaseName, string targetManagedDatabaseId)
+        {
+            GetCurrentSqlClient().ManagedDatabases.StartMove(resourceGroupName, managedInstanceName, databaseName, new ManagedDatabaseStartMoveDefinition()
+            {
+                DestinationManagedDatabaseId = targetManagedDatabaseId,
+                OperationMode = "Move"
+            });
+        }
+
+
+        public void Copy(string resourceGroupName, string managedInstanceName, string databaseName, string targetManagedDatabaseId)
+        {
+            GetCurrentSqlClient().ManagedDatabases.StartMove(resourceGroupName, managedInstanceName, databaseName, new ManagedDatabaseStartMoveDefinition()
+            {
+                DestinationManagedDatabaseId = targetManagedDatabaseId,
+                OperationMode = "Copy"
+            });
+        }
+
+        public void CompleteMoveCopy(string resourceGroupName, string managedInstanceName, string databaseName, string targetManagedDatabaseId)
+        {
+            GetCurrentSqlClient().ManagedDatabases.CompleteMove(resourceGroupName, managedInstanceName, databaseName, new ManagedDatabaseMoveDefinition() 
+            {
+                DestinationManagedDatabaseId = targetManagedDatabaseId
+            });
+        }
+
+        public void CancelMoveCopy(string resourceGroupName, string managedInstanceName, string databaseName, string targetManagedDatabaseId)
+        {
+            GetCurrentSqlClient().ManagedDatabases.CancelMove(resourceGroupName, managedInstanceName, databaseName, new ManagedDatabaseMoveDefinition()
+            {
+                DestinationManagedDatabaseId = targetManagedDatabaseId
+            });
+        }
+
+        public IList<ManagedDatabaseMoveOperationResult> GetMoveOperations(string resourceGroupName, string location, string managedInstanceName, string databaseName, string targetManagedInstanceName, OperationMode mode, bool onlyLatestPerDatabase)
+        {
+            var operationMode = mode == OperationMode.MOVE ? "Move" : "Copy";
+            string filter = $"Properties/OperationMode eq '{operationMode}' and (Properties/SourceManagedInstanceName eq '{managedInstanceName}'";
+
+            if (targetManagedInstanceName.Equals(managedInstanceName))
+            {
+                filter += $" or Properties/TargetManagedInstanceName eq '{targetManagedInstanceName}')";
+            }
```

Comment:
not sure I understand the filtering logic, could you elaborate, why do we have `and` & `or` variants?

Reply:
Thanks for catching this. I have copied same implementation from Portal, but actually here we do not want to do it same.